pkg/generate/app/cmd/newapp.go

```diff
@@ -334,6 +334,14 @@ func (c *AppConfig) addReferenceBuilderComponents(b *app.ReferenceBuilder) {
 		input.Searcher = searcher
 		return input
 	})
+
+	_, repos, _ := b.Result()
+	for _, repo := range repos {
+		repo.SetContextDir(c.ContextDir)
+		if c.Strategy == "docker" {
```

Comment:
actually we don't need it here, inferBuildTypes is sufficient. I was simply repeating the same thing we do for the individual source repos. Will remove.